alien4cloud-tosca/src/main/java/alien4cloud/tosca/serializer/ToscaSerializerUtils.java

```diff
@@ -403,12 +403,13 @@ public static boolean canUseShortNotationForImplementationArtifact(Operation ope
             return Collections.emptyMap();
         }
         // Only generate artifacts that are really stored inside the topology
-        return artifacts.entrySet().stream()
-                .filter(artifact -> (topologyArchiveName.equals(artifact.getValue().getArchiveName())
-                        && topologyArchiveVersion.equals(artifact.getValue().getArchiveVersion()))
-                        || "alien_topology".equals(artifact.getValue().getArtifactRepository())
-                        || (artifact.getValue().getArtifactRef() != null && GET_INPUT_ARTIFACT_PATTERN.matcher(artifact.getValue().getArtifactRef()).matches()))
-                .collect(Collectors.toMap(Map.Entry::getKey, (Map.Entry::getValue)));
+        return artifacts;
```

Comment:
What is this needed for?

Reply:
For the A4C editor. It doesn't load the changes for the artifacts when edited from the UI, so I decided to always load all artifacts and make them available in the editor too. They seem to be used in our DEEP recipes.